src/test/java/org/snomed/snowstorm/rest/ReferenceSetMemberControllerTest.java

```diff
@@ -0,0 +1,82 @@
+package org.snomed.snowstorm.rest;
+
+import org.junit.jupiter.api.Test;
+import org.snomed.snowstorm.AbstractTest;
+import org.snomed.snowstorm.TestConfig;
+import org.snomed.snowstorm.core.data.domain.Concepts;
+import org.snomed.snowstorm.core.data.domain.ReferenceSetMember;
+import org.snomed.snowstorm.core.data.services.ReferenceSetMemberService;
+import org.snomed.snowstorm.loadtest.ItemsPagePojo;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.test.context.ActiveProfiles;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+@ActiveProfiles("test")
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT, classes = TestConfig.class)
+class ReferenceSetMemberControllerTest extends AbstractTest {
+	private static final ParameterizedTypeReference<ItemsPagePojo<ReferenceSetMember>> FIND_REFSET_MEMBERS_RESPONSE_TYPE = new ParameterizedTypeReference<>() {
+	};
+
+	@LocalServerPort
+	private int port;
+
+	@Autowired
+	private TestRestTemplate restTemplate;
+
+	@Autowired
+	private ReferenceSetMemberService referenceSetMemberService;
+
+	@Test
+	void findRefsetMembers_ShouldPage_WhenGivenSearchAfterRequestParameter() throws InterruptedException {
+		// create & assert first reference set member
+		ReferenceSetMember memberA = new ReferenceSetMember(Concepts.MODEL_MODULE, Concepts.REFSET_DESCRIPTOR_REFSET, Concepts.REFSET_MOVED_FROM_ASSOCIATION);
+		referenceSetMemberService.createMember("MAIN", memberA);
+		ItemsPagePojo<ReferenceSetMember> firstRequest = findRefsetMembers();
+		assertEquals(1, firstRequest.getItems().size()); // memberA
+
+		// create & assert second reference set member
+		ReferenceSetMember memberB = new ReferenceSetMember(Concepts.MODEL_MODULE, Concepts.REFSET_DESCRIPTOR_REFSET, Concepts.REFSET_MOVED_TO_ASSOCIATION);
+		referenceSetMemberService.createMember("MAIN", memberB);
+		assertEquals(2, findRefsetMembers().getItems().size()); // memberA & memberB
+
+		// query first page (using offset & limit)
+		ItemsPagePojo<ReferenceSetMember> firstPageMembers = findRefsetMembers(0, 1);
+		assertEquals(1, firstPageMembers.getItems().size());
+		ReferenceSetMember firstPageMember = getReferenceSetMember(firstPageMembers);
+		assertEquals(memberA.getMemberId(), firstPageMember.getMemberId());
+
+		// query second page (using searchAfter token from firstPage)
+		ItemsPagePojo<ReferenceSetMember> secondPageMembers = findRefsetMembers(firstPageMembers.getSearchAfter());
+		Thread.sleep(10000);
```

Comment:
The test would occasionally fail. I added in this sleep which resulted in the test consistently passing. I presume searchAfter is slower? I'm not sure. I've tried decreasing the sleep time but it didn't allow for the test to pass. Do you have any recommendations on a better approach? Sleeping for 10 seconds isn't ideal.